### ISSUE TITLE
Formatter conflict with E501 and comments

### ISSUE THREAD

**Issue by shughes-uk** [astral-sh/ruff#7471]
Ruff version 0.0.289

Before

```python
if (
    "aaaaaaaaaaaaaaaaaaaa"
    and "sdasdaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"  # type: ignore
):
    pass
```

After `ruff format`
```python
if (
    "aaaaaaaaaaaaaaaaaaaa" and "sdasdaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"  # type: ignore
):
    pass
```

There's now an E501 violation, due to the length of the comment. I'm guessing this is a disparity between the formatter ignoring comments for line length calculations and E501 not. 

I understand there's some disparity between the way `black` handles line length and ruff, but i'd expect the ruff formatter to be consistent with ruff's line length rules, which is not the case here.

**Comment by JonathanPlasse** [comment]
Did you try [pycodestyle.ignore-overlong-task-comments](https://docs.astral.sh/ruff/settings/#pycodestyle-ignore-overlong-task-comments) mentioned in [line-too-long](https://docs.astral.sh/ruff/rules/line-too-long/)?

**Comment by charliermarsh** [comment]
Our plan is to modify E501 to allow for overlong pragma comments, to make it consistent with the formatter.

**Comment by charliermarsh** [comment]
(That we allow the formatter to exceed the line length for pragma comments is intended and documented.)

**Comment by gertvdijk** [comment]
Hi @charliermarsh! Just installed 0.0.292, using the Ruff linter (not formatter) and I have found this change to the formatting it wants (removing `# noqa: E501`):

```diff
-    @serial.validator  # pyright: ignore[reportUnknownMemberType, reportUntypedFunctionDecorator, reportGeneralTypeIssues]  # noqa: E501
+    @serial.validator  # pyright: ignore[reportUnknownMemberType, reportUntypedFunctionDecorator, reportGeneralTypeIssues]
```

... which sounds like what's happened in 1646939383336c45f91a553035c6e5b1f1e2341a for this issue.

That change in behaviour for E501 appears to be missing from the changelog, I guess?

```
$ flake8 src/pykmp/messages.py --max-line-length 88
src/pykmp/messages.py:338:89: E501 line too long (122 > 88 characters)
src/pykmp/messages.py:429:89: E501 line too long (125 > 88 characters)
```

This wasn't clear after reading the changelog and I believe deviating from flake8's E501, may deserve a line in the changelog, if that's intentional. If not, I'm happy to create a new issue for this.

**Comment by charliermarsh** [comment]
@gertvdijk - Sorry, that's intentional -- I knew I was missing something from the changelog. I'll update it now.
